scene/animation/animation_mixer.cpp

```diff
@@ -1131,12 +1137,13 @@ void AnimationMixer::blend_capture(double p_delta) {
 
 void AnimationMixer::_blend_calc_total_weight() {
 	for (const AnimationInstance &ai : animation_instances) {
-		Ref<Animation> a = ai.animation_data.animation;
+		const Ref<Animation> &a = ai.animation_data.animation;
 		real_t weight = ai.playback_info.weight;
 		const real_t *track_weights_ptr = ai.playback_info.track_weights.ptr();
 		int track_weights_count = ai.playback_info.track_weights.size();
-		ERR_CONTINUE_EDMSG(!animation_track_num_to_track_cache.has(a), "No animation in cache.");
-		LocalVector<TrackCache *> &track_num_to_track_cache = animation_track_num_to_track_cache[a];
+		AHashMap<Ref<Animation>, LocalVector<AnimationMixer::TrackCache *>>::ConstIterator it = animation_track_num_to_track_cache.find(a);
```

Comment:
What is the benefit of using `find()` and getting the iterator? I suspect that `getptr()` would be faster and it certainly results in easier to read code